File: react-native/library/src/starPlanet/components/TspCheckInStreakCard.js
import React from 'react';
import { Pressable, StyleSheet, Text, View } from 'react-native';
import { withTheme } from '../utils/shared';

/**
 * TspCheckInStreakCard – Continuous check-in summary.
 */
export function TspCheckInStreakCard({
  streakDays = 0,
  totalDays = 0,
  weekProgress = 0,
  disabled = false,
  theme,
  onOpen,
}) {
  const t = withTheme(theme);
  const pct = Math.max(0, Math.min(100, Math.round(Number(weekProgress) * 100)));

  return (
    <Pressable
      disabled={disabled}
      onPress={disabled ? undefined : () => onOpen?.()}
      style={[
        styles.root,
        {
          backgroundColor: t.surfaceRaised,
          borderColor: t.borderDefault,
        },
        disabled && styles.disabled,
      ]}
    >
      <Text style={[styles.title, { color: t.textPrimary }]}>连续打卡</Text>
      <View style={styles.stats}>
        <Text style={[styles.stat, { color: t.textSecondary }]}>{`连续 ${streakDays} 天`}</Text>
        <Text style={[styles.stat, { color: t.textSecondary }]}>{`累计 ${totalDays} 天`}</Text>
        <Text style={[styles.stat, { color: t.textSecondary }]}>{`本周 ${pct}%`}</Text>
      </View>
    </Pressable>
  );
}

const styles = StyleSheet.create({
  root: {
    width: '100%',
    borderWidth: 1,
    borderRadius: 20,
    padding: 16,
    gap: 10,
  },
  title: { fontWeight: '900', fontSize: 16 },
  stats: { flexDirection: 'row', flexWrap: 'wrap', gap: 12 },
  stat: { fontSize: 13, fontWeight: '700' },
  disabled: { opacity: 0.45 },
});

File: flutter/library/lib/src/tsp_check_in_streak_card.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

/// Continuous check-in summary.
class TspCheckInStreakCard extends StatelessWidget {
  const TspCheckInStreakCard({
    super.key,
    this.streakDays = 0,
    this.totalDays = 0,
    this.weekProgress = 0,
    this.disabled = false,
    this.theme = StarPlanetTheme.sky,
    this.onOpen,
  });

  final num streakDays;
  final num totalDays;
  final num weekProgress;
  final bool disabled;
  final StarPlanetTheme theme;
  final VoidCallback? onOpen;

  @override
  Widget build(BuildContext context) {
    final pct = (weekProgress * 100).round().clamp(0, 100);
    return Opacity(
      opacity: disabled ? .45 : 1,
      child: GestureDetector(
        onTap: disabled ? null : onOpen,
        child: Container(
          width: double.infinity,
          padding: const EdgeInsets.all(14),
          decoration: BoxDecoration(
            color: theme.surfaceRaised,
            borderRadius: BorderRadius.circular(18),
            border: Border.all(color: theme.borderDefault),
          ),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: [
              Text('连续打卡', style: TextStyle(color: theme.textPrimary, fontWeight: FontWeight.w900, fontSize: 16)),
              const SizedBox(height: 8),
              Wrap(
                spacing: 16,
                runSpacing: 8,
                children: [
                  Text('连续 $streakDays 天', style: TextStyle(color: theme.textSecondary, fontWeight: FontWeight.w700)),
                  Text('累计 $totalDays 天', style: TextStyle(color: theme.textSecondary, fontWeight: FontWeight.w700)),
                  Text('本周 $pct%', style: TextStyle(color: theme.textSecondary, fontWeight: FontWeight.w700)),
                ],
              ),
            ],
          ),
        ),
      ),
    );
  }
}
